src/drm/fairplay.js

```diff
@@ -28,11 +30,13 @@ export default class FairPlay extends BaseDrmProtocol {
    * Sets the FairPlay playback.
    * @param {HTMLVideoElement} videoElement - The video element to manipulate.
    * @param {Array<Object>} drmData - The drm data.
+   * @param {any} errorCallback - player error object
    * @returns {void}
    */
-  static setDrmPlayback(videoElement: HTMLVideoElement, drmData: Array<Object> = []): void {
+  static setDrmPlayback(videoElement: HTMLVideoElement, drmData: Array<Object> = [], errorCallback: any): void {
```

Comment:
@OrenMe 
maybe to avoid the callback pattern, we can trigger the error from the video element, i.e. `videoElement.dispatchEvent(....)`. it will be catched by design in the html5 engine

Reply:
@dan-ziv IMHO it's not a best practice to use the video to fire an event which is not of his domain.
Passing the error handle as delegate is OK, and in general I think the DRM flow should be promise based(as the standard EME is) so we can just `throw` on exception/error and then bind the promise catch in the higher level and emit the error event there - but this is another refactor